elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/MappingDsl.scala

```diff
@@ -39,11 +39,15 @@ trait MappingDsl extends DslCommons {
   sealed trait IndexType {
     val rep: String
   }
-
+  
   case object NotAnalyzedIndex extends IndexType {
     val rep = "not_analyzed"
   }
 
+  case object NotIndexedIndex extends IndexType {
```

Comment:
A comment explaining why someone might want this seemingly nonsensical mode or a link to the docs could be useful.